client/components/edit/children/ChildPosition.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from "react";
+import TextField from "@mui/material/TextField";
+import { useEditorContext } from "../../../context/EditorContext";
+import { useFetchContext } from "../../../context/FetchContext";
+import { getPositionInParentUrl } from "../../../util/routes";
+
+export interface ChildPositionProps {
+    pid: string;
+    parentPid: string;
+}
+
+export const ChildPosition = ({ pid, parentPid }: ChildPositionProps): React.ReactElement => {
+    const {
+        state: { objectDetailsStorage },
+        action: { clearPidFromChildListStorage, removeFromObjectDetailsStorage },
+    } = useEditorContext();
+    const {
+        action: { fetchText },
+    } = useFetchContext();
+    const [statusMessage, setStatusMessage] = useState<string>("");
+
+    // Important: this component assumes it will be called in a context where pid and parent pid
+    // details are already loaded into the context; it will not cause a load to occur by itself.
+    const loaded = Object.prototype.hasOwnProperty.call(objectDetailsStorage, pid);
```

Comment:
Before calling this component, can you check pid exists before displaying it? Or do you want it to partially render?

Reply:
You're right that we can avoid rendering this component in some situations. I've made it conditional with https://github.com/FalveyLibraryTechnology/VuDLPrepJS/pull/147/commits/38b614db254815a5b142966680ee08e805327fa7. I think there's still some value in keeping the loaded checks in the component itself, just in case it gets used in a different context -- this makes it slightly more robust.